test/requests/definition_expectations_test.rb

```diff
@@ -93,4 +102,33 @@ def test_jumping_to_default_gems
   ensure
     T.must(message_queue).close
   end
+
+  def test_jumping_to_default_require_of_a_gem
```

Comment:
Let's add a test showing the current behavior when jumping to `lib/foo` with the following entries:

* `lib/foo.rb`
* `lib/foo/bar.rb`
* `lib/foo/baz.rb`